python_modules/dagster/dagster/_core/blueprints/blueprint.py

```diff
@@ -0,0 +1,158 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable, Mapping, NamedTuple, Optional, Union
+
+from dagster import _check as check
+from dagster._core.definitions.asset_checks import AssetChecksDefinition
+from dagster._core.definitions.assets import AssetsDefinition, SourceAsset
+from dagster._core.definitions.cacheable_assets import CacheableAssetsDefinition
+from dagster._core.definitions.definitions_class import Definitions
+from dagster._core.definitions.executor_definition import ExecutorDefinition
+from dagster._core.definitions.job_definition import JobDefinition
+from dagster._core.definitions.logger_definition import LoggerDefinition
+from dagster._core.definitions.partitioned_schedule import (
+    UnresolvedPartitionedAssetScheduleDefinition,
+)
+from dagster._core.definitions.schedule_definition import ScheduleDefinition
+from dagster._core.definitions.sensor_definition import SensorDefinition
+from dagster._core.definitions.unresolved_asset_job_definition import UnresolvedAssetJobDefinition
+from dagster._core.errors import DagsterInvariantViolationError
+from dagster._core.executor.base import Executor
+from dagster._model import DagsterModel
+from dagster._utils.source_position import HasSourcePositionAndKeyPath
+
+
+class DagsterBuildDefinitionsFromConfigError(Exception):
+    pass
+
+
+class BlueprintDefinitions(NamedTuple):
```

Comment:
`Definitions` feels like a first-class noun that at first glance I would expect `BlueprintDefinitions` to implement. I guess it's using it in the lowercase-d "definitions" sense, but this threw me for a bit of a loop.

How would you feel about e.g. `BlueprintDefs` or even `PartialDefinitions`, `DefinitionsFragment` (leaving open the door for non-Blueprint use-case)?

Is there a world in which we'd have some sort of definitions ABC that would let you `.merge` a `BlueprintDefinitions` with `Definitions`?

Reply:
Is `BlueprintDefinitions` replacing the concept of merging/joining `Defintions`? (eg are we returning to the world in which `Definitions` is singular)?

Reply:
Chatted this through in person, but for posterity:
- I hope to eliminate `BlueprintDefinitions` and replace it with `Definitions`
- Using `Definitions` here will require making it possible to construct a `Definitions` object without binding all the resource keys to resource definitions, which is a behavior change from the current `Definitions` object. If we do make that change, we will probably want to at least wait until 1.8.  Adding a new class allows us to make progress on blueprints without waiting. More discussion here: https://github.com/dagster-io/dagster/pull/21746.